src/kinds/extractTable.ts

```diff
@@ -64,6 +65,27 @@ export interface TableColumn {
   informationSchemaValue: InformationSchemaColumn;
 }
 
+export type TableIndexColumn = {
+  /**
+   * Column name or null if functional index
+   */
+  name: string | null;
+  /**
+   * Definition of index column
+   */
+  definition: string;
+};
+
+export type TableIndex = {
```

Comment:
Would `interface` be preferable to `type`?

After implementing this, I noticed that the type is not rendered as an interface in the API documentation. The api-documenter does not seem to render the type properties as plainly as interface.
https://github.com/microsoft/rushstack/issues/1807

Reply:
Yeah, I remember this as well. The API documenter is not great IMO, but it was the best I could find at the time. 

I originally preferred `type` over `interface` because the former felt like the more Typescript-native way and interfaces seemed like something that belongs to Java or C#. However, they do have one other advantage: better error messages. The `type` keyword just works as a sort of macro and when an error is encountered, TSC will show you the whole expanded type. On the other hand, a missing property in an index will show up just as that. 

So, yes, I think interfaces might be preferable :-)